evaluation.md

```diff
@@ -5,6 +5,27 @@
 _Describe each change from your ideal syntax to the syntax you implemented, and
 describe_ why _you made the change._
 
+We got into the IDE and immediately realized that things were going to be harder than we thought they would be. We  went through various discussions of what would be objects,classes, and operators and eventually settled on a compromise of where to start. We changed to:
+
+'rule( # seq ) -> action' which is pretty different but we're planning to slim down significantly by removing the need for the rule keyword.
+
+Yikes. We're figuring out how we want to pass our arguments to rule and so far the only way we can see to do that is as a list. This isn't ideal but hopefully we'll be able to make it better. End day 1
+
+We realized that using a list would require users to use nested parentheses when writing the syntax, so we decided to switch to `rule` being a class whose constructor accepts an int representing the state, followed by *args of position statuses (i.e., whether the north wall is open or blocked, etc.).
+
+We remembered currying! wow! this changes everything.
+
+Wow this is super cool. We are feeling significantly less scared and significantly more excited about this lab and also about scala. Robin especially thinks that scala is awesome.
+
+We have come a long way about our compromises. At first we were hoping to compromise not very much and then about 1.5 hours in we were thinking that everything would have to be completely compromised and now were striking a balance!
+
+We're done, yay! Now that we have hindsight we think that it still feels pretty scala-y which is kind of a bummer. This is mostly becuse we have to pass our rules as a list and also the parentheses really suck. Also all the imports and stuff at the bottom is kind of annoying but we think that if we were to render it as it is rendered in picobot that we could hide a lot of that stuff.
```

Comment:
As I said above, we ignored the boilerplate code. However, you implemented this differently (by extending `JFXApp`), so I think you could have handled it easily. If you’d included a `val rules: Rule` and a `val maze: Maze` constructor parameter in your trait, then you could have moved that boilerplate to the trait instead. This would also have gotten rid of the need for most of the imports for people writing Picobot code.